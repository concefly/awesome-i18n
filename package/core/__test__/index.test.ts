import AwesomeI18n from '../lib/index';
import MemoryFileSystem = require('memory-fs');
import { BaseTranslator, ITranslateOpt } from 'ai18n-type';

const defaultConfig = {
  input: '.',
  output: './i18n',
  defaultLang: 'zh-cn',
  langs: ['zh-cn', 'en'],
  logger: null,
  loader: [
    {
      test: '\\.tsx?$',
      use: 'ai18n-loader-ts',
    },
    {
      test: '\\.jsx?$',
      use: 'ai18n-loader-ts',
    },
  ],
  reducer: 'ai18n-reducer',
  translator: 'ai18n-translator-google',
};

class FakeTranslator extends BaseTranslator {
  async translate(text: string, opt: ITranslateOpt) {
    return { message: `${text}_${opt.from}_${opt.to}` };
  }
}

test('normal', async () => {
  const mfs = new MemoryFileSystem();

  mfs.writeFileSync('/a.js', `<div>{__('苹果')}</div>`, 'utf-8');
  mfs.writeFileSync('/b.js', `<div>{__define('香蕉')}</div>`, 'utf-8');
  mfs.writeFileSync('/c.tsx', `<div>{__define('西瓜')}</div>`, 'utf-8');

  mfs.mkdirSync('/i18n');
  mfs.writeFileSync('/i18n/en.json', JSON.stringify({ 苹果: 'apple' }), 'utf-8');

  const i18n = new (class extends AwesomeI18n {
    getInputFiles() {
      return ['/a.js', '/b.js', '/c.tsx'];
    }
    getFs() {
      return mfs as any;
    }
  })({
    ...defaultConfig,
    output: '/i18n',
    translator: new FakeTranslator(),
  });
  const result = await i18n.run();
  expect(result).toMatchSnapshot();
});
