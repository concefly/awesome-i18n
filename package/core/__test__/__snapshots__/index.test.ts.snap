// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal 1`] = `
Object {
  "en": Object {
    "苹果": "apple",
    "西瓜": "西瓜_zh-cn_en",
    "香蕉": "香蕉_zh-cn_en",
  },
  "zh-cn": Object {
    "苹果": "苹果",
    "西瓜": "西瓜",
    "香蕉": "香蕉",
  },
}
`;